### PR TITLE
Oo service monitors

### PR DESCRIPTION
**Pull request by laurafitzgerald**
# Issue link
Replaces https://github.com/valerymo/integreatly-operator/pull/29

# What

# Verification steps
Run this branch with the following steps
INSTALLATION_TYPE=managed-api make cluster/prepare/local
INSTALLATION_TYPE=managed-api make code/run
You may need to do the image pull workaround

Verify that there are Service Monitors present in the observability prometheus 
There is a related pr which needs to merge -> https://github.com/redhat-developer/observability-operator/pull/89

### CODE REVIEW

File: pkg/products/observability/reconciler.go [hunk 1]
```diff
@@ -325,6 +324,7 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient k8scl
 	}
 
 	op, err := controllerutil.CreateOrUpdate(ctx, serverClient, oo, func() error {
+		overrideSelectors := true
 		disabled := true
 		oo.Spec = observability.ObservabilitySpec{
```
[the remark this answers is not in the document]
Comment (laurafitzgerald) [reply pkg/products/observability/reconciler.go:329]: Yes, it's a good point. we can never manually add something to the spec if needed. 
This pr needs a rebase again. let me see if I can improve this.

**Comment by Boomatang** [comment]
Rechecked after the force push. All the changes are still working as expected.
/lgtm

**Comment by Boomatang** [comment]
Service monitors are showing up.
/lgtm

File: pkg/config/observability.go [hunk 2]
```diff
@@ -108,16 +113,32 @@ func (m *Observability) GetAlertManagerVersion() string {
...
+	return AlertManagerOverride
+}
+
 func (m *Observability) GetPrometheusVersion() string {
 	return "v2.29.2"
 }
 
 func (m *Observability) GetPrometheusRouteName() string {
-	return "prometheus-route"
+	return PrometheusOverride
```
Comment (KevFan) [inline pkg/config/observability.go:133]: This will affect test cases / scripts referencing the old route name

File: pkg/config/observability.go [hunk 3]
```diff
@@ -108,16 +113,32 @@ func (m *Observability) GetAlertManagerVersion() string {
...
 }
 
-func (m *Observability) GetAlertManagerRouteName() string {
-	return "alertmanager-route"
```
Comment (KevFan) [inline pkg/config/observability.go:120]: This will affect test cases / scripts referencing the old route name

**Pull request merged**